mitodl/open-discussions
# Display similar learning resources in drawer

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #2436 

#### What's this PR do?
- Adds a backend API to search for similar resources and adjusts the UI to display them.

#### How should this be manually tested?
- Open a drawer for any resource, from either the course search page or home page.  At the bottom there should be a 'Similar Learning Resources' section with a '+' next to it.
- Click the '+', it should change to a '-' and the similar resources should be displayed.
- Click the '-' it should change to a '+' and the similar resources should be hidden again.

#### Any background context you want to provide?
Initially I tried sending `object_id` and `object_type` to the API as parameters, then using `like={"_id": gen_doc_id(object_id, object_type)}` for the ES 'More Like This' query, but this only returned resources of the same type.  

So I switched to sending a document body in the request containing the following fields -  `id`, `object_type`, `title`, `short_description` - and using `like={"doc": value_doc}` instead.  Another option would be to continue sending object id and type, retrieve and serialize the correct object from that, and use the serializer data as the `value_doc`.  But that's an additional database call so I decided to not do it this way.

#### Screenshots (if appropriate)
(Optional)
